python/langsmith/evaluation/llm_evaluator.py

```diff
@@ -0,0 +1,221 @@
+            prompt. Defaults to None. If None, it is assumed that the prompt
+                only requires 'input', 'output', and 'expected'.
+            model (Optional[str], optional): The model to use for the evaluation.
+                Defaults to "gpt-3.5-turbo".
+            model_provider (Optional[str], optional): The model provider to use
+                for the evaluation. Defaults to "openai".
+        """
+        try:
+            from langchain_core.prompts import ChatPromptTemplate
+        except ImportError as e:
+            raise ImportError(
+                "LLMEvaluator requires langchain-core to be installed. "
+                "Please install langchain-core by running `pip install langchain-core`."
+            ) from e
+        try:
+            from langchain.chat_models import init_chat_model
+        except ImportError as e:
+            raise ImportError(
+                "LLMEvaluator requires langchain to be installed. "
+                "Please install langchain by running `pip install langchain`."
+            ) from e
+        if isinstance(prompt_template, str):
+            self.prompt = ChatPromptTemplate.from_messages(
+                [("system", prompt_template)]
+            )
+        else:
+            self.prompt = ChatPromptTemplate.from_messages(prompt_template)
+
+        if set(self.prompt.input_variables) - {"input", "output", "expected"}:
+            if not map_variables:
+                raise ValueError(
+                    "map_inputs must be provided if the prompt template contains "
+                    "variables other than 'input', 'output', and 'expected'"
+                )
+        self.map_variables = map_variables
+
+        self.score_config = score_config
+        self.score_schema = _create_score_json_schema(self.score_config)
+
+        try:
+            chat_model = init_chat_model(
+                model=model, model_provider=model_provider, **kwargs
+            ).with_structured_output(self.score_schema)
+        except ImportError as e:
+            raise ImportError(
+                "LLMEvaluator is missing a required langchain integration."
+            ) from e
+        except ValueError as e:
+            raise ValueError(
+                "Error loading the model. Please check the model, model_provider, "
+                "and that the appropriate secrets are set."
+            ) from e
+
+        self.runnable = self.prompt | chat_model
+
+    def evaluate_run(
+        self, run: Run, example: Optional[Example] = None
+    ) -> Union[EvaluationResult, EvaluationResults]:
+        """Evaluate a run."""
+        if self.map_variables:
+            variables = self.map_variables(run, example)
+            if set(self.prompt.input_variables) - set(variables.keys()):
+                raise ValueError(
+                    "map_variables must return a dictionary with keys for all of the "
```

Comment:
Wouldn't we get this for free when we invoke the model?

Reply:
yeah but thought it would be better to fail fast at construction